### ISSUE TITLE
No way to disable Dumpling on Helix runs

### ISSUE THREAD

**Issue by MattGal** [dotnet/buildtools#1414]
See https://github.com/dotnet/buildtools/blob/66dc7235af5e0a9f3edcaa929da7f0a5fdd21259/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L343

We should add some sort of explicit check for false, or any other way to opt out of this.
@sepidehMS  FYI.

**Comment by sepidehMS** [comment]
+ @mellinoe

**Comment by mellinoe** [comment]
> We should add some sort of explicit check for false

That sounds reasonable to me.

**Comment by sepidehMS** [comment]
@MattGal I agree that Dumpling should not be enabled by only `'$(EnableCloudTest)' == 'true'` in [here](https://github.com/dotnet/buildtools/blob/66dc7235af5e0a9f3edcaa929da7f0a5fdd21259/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L343). But just to get a clearer picture, how would you opt out of this if you wanted to? Would that be by manually editing the build configurations somehow? 
I'm asking because I believe the correct way would be to add `/p:EnableDumpling=true"` to pipeline.json for all Helix builds like [this](https://github.com/dotnet/corefx/blob/master/buildpipeline/pipeline.json#L515) is doing for Linux. But I'm not sure what's the best way to opt out of that.
@mellinoe Considering pipeline.json doesn't pass in `/p:EnableDumpling=true"` for Windows and OSX builds, how was Dumpling enabled for those on Helix runs?

**Comment by MattGal** [comment]
@sepidehMS the problem is this condition turns it on regardless of the value of EnableDumpling: Condition="'$(EnableDumpling)' == 'true' OR '$(EnableCloudTest)' == 'true'"   so I could explicitly say:

`  /p:EnableDumpling=false /p:EnableCloudTest=true `

... and still get the Dumpling enabled.   Lots of simple ways to fix this, was just raising awareness.

**Comment by sepidehMS** [comment]
@MattGal I agree that's a problem to fix. My question was if we change this condition and only rely on and check for `/p:EnableDumpling=true` when importing Dumpling.targets, would you be able to opt out of this for random Helix runs even if pipeline.json passes  `/p:EnableDumpling=true` to all Helix builds?

**Comment by MattGal** [comment]
it's up to you, my thought would be to conditionally set it if it's not already set, i.e. in a property group:

`<EnableDumpling Condition="'$(EnableDumpling)'=='' and '$(EnableCloudTest)'=='true'>true</EnableDumpling>`

This gets you your old behavior of turning on when Cloud test is true, but honors the scenario above.

**Comment by ViktorHofer** [comment]
// Auto-generated message

Going forward, the .NET team is using https://github.com/dotnet/arcade to develop the code and issues formerly in this repository. Feel free to reopen/move this issue if it still applies to servicing branches in this repository, or source code which now resides in the Arcade repository.
